test/jdk/javax/swing/JComponent/7154030/bug7154030.java

```diff
@@ -81,25 +84,36 @@ public void run() {
                     button.setOpaque(true);
                     button.setVisible(false);
                     desktop.add(button);
+                    desktop.setMinimumSize(new Dimension(300, 300));
+                    desktop.setMaximumSize(new Dimension(300, 300));
 
                     frame.setContentPane(desktop);
-                    frame.setSize(300, 300);
+                    frame.setMinimumSize(new Dimension(350, 350));
+                    frame.setMaximumSize(new Dimension(350, 350));
+                    frame.pack();
```

Comment:
Wouldn't it make {{desktop}} larger than 300,300?

Reply:
Yes, that was a preliminary version, i forgot to commit the final version before creating the pull request. Fixed.

Reply:
It will but it forces the invalidation and repainting of the content so test does not fail.

Reply:
> It will but it forces the invalidation and repainting of the content so test does not fail.

Got it! It's the intention to cause additional repaint.